refactor/results/app/api/api_v0/endpoints/workflow.py

```diff
@@ -217,3 +217,15 @@ def update_result(*, dispatch_id: str, task: Node) -> Any:
     pickle.dump(results_object, pickled_result)
     if _upload_file(pickled_result):
         return {"response": "Task updated successfully"}
+
+
+@router.delete("/results/{dispatch_id}", status_code=200)
+def delete_result(*, dispatch_id: str) -> Any:
+    # Retrieve file path from db and request deletion from data service
+    filename = _get_result_from_db(dispatch_id, "filename")
+    r = requests.delete(f"http://{base_url}/delete", params={"obj_name": filename})
+
+    sql = "DELETE FROM results WHERE dispatch_id = ?"
```

Comment:
This should only delete from the DB if the object deletion was successful. Otherwise we will create orphaned objects.